src/core/hrs.ml

```diff
@@ -110,8 +115,11 @@ let to_HRS : Format.formatter -> Sign.t -> unit = fun oc sign ->
   Format.fprintf oc "\n(COMMENT rewriting rules)\n";
   let print_rules s =
     match !(s.sym_def) with
-    | Some(d) -> Format.fprintf oc "(RULES %a\n" print_sym s;
-                 Format.fprintf oc "    -> %a)\n" (print_term false) d
-    | None    -> List.iter (print_rule oc s) !(s.sym_rules)
+    | Some(d) -> print_rule oc (Symb(s,Nothing)) d
+    | None    ->
+      let rules = !(s.sym_rules) in
+      let lhs r = Basics.add_args (Symb(s,Qualified)) r.lhs in
```

Comment:
Why do you use Qualified here, and Nothing before? Why not using Nothing everywhere, and thus replace `Symb(s,Qualified)` by  `symb s` again here?